### ISSUE TITLE
Session should not implement Send (at least if there is no mutual exclusion)

### ISSUE THREAD

**Issue by ubnt-intrepid** [alexcrichton/ssh2-rs#154]
According to `libssh2`, the pointer to the raw session handle (`*mut LIBSSH2_SESSION`) will be copied when the other libssh2 types such as `LIBSSH2_CHANNEL` are created, and its pointer will be used when the APIs for these types are called. For example, `libssh2_channel_read_ex` will perform the read operation on the underlying transport inside it<sup>[1]</sup>. These functions are not guaranteed the thread safety and hence the binding library must ensure that all pointer values spawned from a `LIBSSH2_SESSION` remain a single thread, unless the proper mutual exclusion control is used.

However, by implementing the *forced* `Send` implementation for `Session`, the following problematic code where the raw session handle is shared across the multiple threads without any mutual exclusion controls passes the compiler check:

```rust
let mut sess = Session::new().unwrap();
sess.set_tcp_stream(TcpStream::connect("127.0.0.1:22").unwrap());
sess.handshake().unwrap();
...

let mut channel1 = session.channel_session().unwrap();

std::thread::spawn(move || {
    // `session` can be moved since it is `Send`.

    let mut channel2 = session.channel_session().unwrap();
    channel2.exec("ls").unwrap();
    ...
});

channel1.exec("pwd").unwrap();
```

In the above example, the data race may occur when `channel1.exec` and `channel2.exec` are executed at the same time. In order to prevent such cases, it is necessary to remove the `Send` implementation from `Session` that is currently forcibly implemented.

[1]: https://github.com/libssh2/libssh2/blob/f1b6fca89b2c3ca5ea0f47290ad7010d4381bbad/src/channel.c#L2017

**Comment by wez** [comment]
Fixed in 0.7.0 on crates.io

**Comment by ubnt-intrepid** [comment]
Thank you for the quick fix!
